test/release/examples/primers/loops.chpl

```diff
@@ -0,0 +1,950 @@
+  the available processor cores on each of the locales that own a
+  portion of ``C`` since ``C`` is the first iterand in the `zip`
+  expression..
+  
+
+  A final note on data-parallel loops and legality / races
+  --------------------------------------------------------
+
+  As mentioned previously, the Chapel compiler and language are not
+  responsible for making sure that a data-parallel loop is safe to
+  execute in parallel.  If the programmer writes a data-parallel loop
+  that is not parallel-safe or that creates a race, the outcome is
+  their responsibility rather than Chapel's.  As an example, the
+  following loop may appear to replace the interior elements of an
+  array with the average of their neighbors; yet, because the same
+  elements may be read and written simultaneously by distinct parallel
+  iterations, the results will be unpredictable depending on how the
+  iterations are scheduled at execution time:
+
+*/
+
+  var D = [i in 1..n] (i**2): real;
+
+  writeln("Before the race-y averaging loop, D is: ", D);
+  forall i in 2..<n do
+    D[i] = (D[i-1] + D[i+1]) / 2;
+
+  // if the following line were uncommented, you would likely see
+  // different results after each execution of the program:
+  //
+  // writeln("After the race-y averaging loop, D is: ", D);
+
+/*
+
+  The programmer is still permitted to write such loops in Chapel, and
+  the compiler will dutifully implement them as requested; but it will
+  not protect the user from such races.
+
+  One way to write the averaging computation above in a correct, but
+  still parallel, manner would be to store the results into a distinct
+  array to avoid reading and writing and writing the same elements
+  within a single parallel loop:
+
+*/
+
+  var E = [i in 1..n] (i**2): real;
+  var F: [1..n] real;
+
+  writeln("Before the safe averaging loop, E is: ", E);
+
+  forall i in 2..<n do
+    F[i] = (E[i-1] + E[i+1]) / 2;
+
+  writeln("After the safe averaging loop, F is: ", F);
+
+/* 
+
+  Part of the reason the Chapel compiler does not prevent writing
+  parallel loops with races is that it can be difficult to determine
+  whether a given loop is safe to parallelize or not.  For example,
+  the following variation on the original loop would be safe since it
+  only writes to even elements and reads from odd ones:
+
+*/
+
+  D = [i in 1..n] (i**2): real;
+
+  writeln("Before the third averaging loop, D is: ", D);
+  forall i in 2..<n by 2 do
+    D[i] = (D[i-1] + D[i+1]) / 2;
+
+  writeln("After the third averaging loop, D is: ", D);
+
+/*
+
+  Distinguishing between loops that are parallel-safe versus not is
+  generally intractable, so rather than attempting to make that
+  judgment, Chapel trusts the programmer to use the loop form they
+  want.  Moreover, for some parallel computations, race conditions can
+  be benign or acceptable, so Chapel does not want to prevent a user
+  from writing such computations.
```

Comment:
You could compare this to OpenMP and/or other programming models that provide similar functionality.